### ISSUE TITLE
Method mark_hash_map_values() invokes Ruby code; which can crash the Ruby process

### ISSUE THREAD

**Issue by charlez** [rdp/google_hash#39]
Running multi-threaded Ruby applications that make use of google_hash occasionally crash the Ruby process during garbage collection.  Method mark_hash_map_values() uses erb (and hence invokes Ruby), and should not do so:

> Please refer to: https://bugs.ruby-lang.org/issues/11799
> In particular, note Koichi Sasada's comments in #5 and #6.

**Comment by rdp** [comment]
Wow good find.  I have attempted to address it in v 0.9.0 care to test it
for me?

**Comment by charlez** [comment]
Yes, be glad to test it, and report the results.  I'll push your request into my work queue.
